php/Terminus/Commands/TerminusCommand.php

```diff
@@ -34,20 +31,30 @@
   protected $outputter;
 
   /**
-   * @var Request
+   * @var Session
    */
-  protected $request;
+  protected $session;
+
+  /**
+   * @var Sites
+   */
+  protected $sites;
 
   /**
    * Instantiates object, sets cache and session
+   *
+   * @param array $dependencies Elements as follow:
+   *        FileCache cache
+   *        Logger    Logger
+   *        Outputter Outputter
+   *        Session   Session
+   * @return TerminusCommand
    */
-  public function __construct() {
-    //Load commonly used data from cache
-    $this->cache     = Terminus::getCache();
-    $this->logger    = Terminus::getLogger();
-    $this->outputter = Terminus::getOutputter();
-    $this->session   = Session::instance();
-    $this->request   = new Request();
+  public function __construct(array $dependencies = []) {
```

Comment:
can we call this `$options` and have it explicitly set them in the constructor? That way it's explicit what is required and we can error in the constructor if they're missing rather than when a method tries to access the property and it's null.